pkg/router/template/template_helper.go

```diff
@@ -170,13 +170,55 @@ func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
 	return endpoints
 }
 
+// Returns weight for the endpoint.
+// svcWeight - total service weight
+// serviceEndpoints - number of endpoints in service
+// idx - which endpoint is being calculated
+//
+// svcWeight is distributed among the number of serviceEndpoints.
+// Each endpoint gets a fraction of the service's weight, and the
+// sum of the endpoint weights is the service weight.
+// Because the serviceEndpoints may not divide evenly into svcWeight
+// one endpoint may get less than its share.
+// Also, since the minimum weight is 1, there may be more than
+// svcWeight serviceEndpoints so the extra endpoints will get 0 weight.
+func calcWeight(svcWeight, serviceEndpoints int32, idx int) int32 {
```

Comment:
Can we choose between the existing algorithm and this one?

Reply:
I prefer not to have an option to choose. The current quick and dirty algorithm is broken. The new algorithm is also broken. I will re-work this.